sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1144,18 +1144,20 @@ class Analyzer(override val catalogManager: CatalogManager)
         lookupRelation(u.multipartIdentifier, u.options, u.isStreaming)
           .map(resolveViews).getOrElse(u)
 
-      case u @ UnresolvedTable(identifier, cmd) =>
+      case u @ UnresolvedTable(identifier, cmd, relationTypeMismatchHint) =>
         lookupTableOrView(identifier).map {
-          case v: ResolvedView => throw QueryCompilationErrors.expectTableNotViewError(v, cmd, u)
+          case v: ResolvedView =>
```

Comment:
@cloud-fan looks like I missed handling this in `ResolveTempViews`. I will create a follow up PR. Sorry about that.